### DISCUSSION TITLE
Preventing Direct Access to Images Stored on Minio Server

### DISCUSSION THREAD

**Discussion by surajchhetry** [minio/minio#17546] [Q&A] [answered]
Hello everyone,

I am currently using Minio Server to store images for a web application. The application retrieves and displays these images via presigned URLs, which works perfectly fine.

However, I've encountered an  issue: users are able to copy the presigned URL and paste it into a new browser tab. This allows them to directly access and view the image outside the application, which isn't the desired behavior. The goal is to ensure that the images can only be viewed within the context of the web application.

One possible solution I've considered is configuring cross-origin settings in Minio. But before implementing it, I wanted to gather some insights from the community:

    1. Can this issue be effectively resolved by configuring cross-origin settings in Minio?
    2. Are there alternative or better solutions to prevent direct access to the images using the presigned URL?

I appreciate any suggestions or guidance on how to address this situation. 

Thank you for your time and expertise!

**Comment by harshavardhana** [comment, marked as answer]
For that you need to generate presigned URLs that terminate on your web app and translate them back to MinIO for access. 

This is not a security issue just that you do not understand how presigned URLs work.

**Reply by surajchhetry** [reply]
It means image need to be download by web app and then send to the client right ?
